tests/test_flow/test_persistence_fuzz.py

```diff
@@ -221,14 +248,23 @@ def run(self, n_iterations):
                     returned_value = self.model.build_flow().get(affected_name)
                     expected_value = self.model.expected_entity_value(
                         affected_name)
-                    if make_func_change:
+
+                    # value will be different if we made a functional change and
+                    # updated entity does not change per run. If updated entity does
+                    # change, the value hash will change and everything will recompute.
+                    if make_func_change and not is_updated_entity_nondeterministic:
                         assert returned_value != expected_value
                     else:
                         assert returned_value == expected_value
 
-                    # We should have used the cached value, so no new
-                    # computation should have happened.
-                    assert len(self.model.called_entity_names()) == 0
+                    # If updated entity does not change per run, we call only nondeterministic
```

Comment:
I'm not sure what the `It's safe ...` part means: could you elaborate on why it wouldn't be safe and what the "later check" refers to?

Reply:
Fair. This is one of the trickiest part of the test now IMO.